test/test_cuda.py

```diff
@@ -2450,7 +2450,12 @@ def _grad_scaling_autocast_fused_optimizers(self, optimizer_ctor, optimizer_kwar
                     actual = state_scaling[k]
                     if k == "step":
                         actual = actual.squeeze()
-                    self.assertEqual(state_control[k], actual, msg=k)
+
+                    expected = state_control[k]
+                    if isinstance(state_control[k], torch.Tensor) and state_control[k].dtype == torch.float64:
+                        expected = torch._cast_Float(state_control[k])
```

Comment:
What is this function?

Reply:
Yeah I just looked up the proper way of casting